ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -425,24 +414,57 @@ impl<'a> Assemble<'a> {
             #[cfg(debug_assertions)]
             infos.sort_by(|a, b| a.0.cmp(&b.0));
 
+            let deopt_label = self.asm.new_dynamic_label();
             let guardcheck_label = self.asm.new_dynamic_label();
             for (deoptid, fail_label) in infos {
                 self.comment(
                     self.asm.offset(),
-                    format!("Deopt ID for guard {:?}", deoptid),
+                    format!("Deopt ID and patch point for guard {:?}", deoptid),
                 );
                 // FIXME: Why are `deoptid`s 64 bit? We're not going to have that many guards!
+
+                // Word-align this location so we can patch this later in a single atomic
+                // operation.
+                let off = self.asm.offset();
+                let align = 8 - (off.0 % 8);
```

Comment:
If we are already aligned, doesn't this push 8 superfluous nops?

Reply:
Since we're on nightly, we can use the `checked_next_multiple_of` function instead of calculating alignment ourselves. [As Edd points out, it's surprisingly difficult to do this correctly!]

Reply:
Oh, you're right. Gonna look into `checked_next_multiple_of`.

Reply:
[096e8be](https://github.com/ykjit/yk/pull/1591/commits/096e8be03a2fc213ff50e2e19c0a08d7497ecd7c)